File: week-2/js/todo-list/src/createApp.js
const express = require('express')


function createApp() {
    const app = express()
    app.use(express.json());
    const todos = [];


    app.get('/todos', (req, res) => {
        res.json(todos);
    })


    app.post('/todos', (req, res) => {
        const newItem = {
            id: Date.now(),
            description: req.body.description,
            status: 'Planned'
        }
        todos.push(newItem)
        res.status(201).json(newItem)
    })



    return app;
}

module.exports = createApp

File: week-2/python/todo-list/src/create_app.py
from flask import Flask, jsonify, request
import time

def create_app():
    app = Flask(__name__)
    todos = []

    @app.get("/todos")
    def get_todos():
        '''
            Return the list of todos
        '''
        return jsonify(todos)
    
    
    @app.post("/todos")
    def add_todo():
        body = request.get_json()
        
        if not isinstance(body['description'], str):
            return jsonify({'error': "You cannot create todos with numbers"}), 400
        
        new_item = {
            "id": int(time.time()*100),
            "description": body['description'],
            "status": "Planned"
        }
        todos.append(new_item)
        return jsonify (new_item), 201

    @app.patch("/todos/<id>")
    def update_todo(id):
        body = request.get_json()
        id = int(id)
        for item in todos:
            if item['id'] == id:
                item['status'] = body['status']
                return jsonify(item)
        
        return jsonify({"error": "Item does not exist"}), 404        
    
    return app
